src/PerfCounters_x86.h

```diff
@@ -265,11 +265,46 @@ static void check_for_xen_pmi_bug() {
   }
 }
 
+static void check_for_zen_speclockmap() {
+  // When the SpecLockMap optimization is not disabled, rr may not work
+  // reliably (e.g. it would work fine on a single process with a single
+  // thread, but not more). When the optimization is disabled, the
+  // perf counter for retired lock instructions of type SpecLockMapCommit
+  // (on PMC 0x25) stays at 0.
+  // See more details at https://github.com/mozilla/rr/issues/2034.
+  struct perf_event_attr attr;
+  // 0x25 == RETIRED_LOCK_INSTRUCTIONS - Counts the number of retired locked instructions
+  // + 0x08 == SPECLOCKMAPCOMMIT
+  init_perf_event_attr(&attr, PERF_TYPE_RAW, 0x510825);
+
+  ScopedFd fd = start_counter(0, -1, &attr);
+  if (fd.is_open()) {
+    int atomic = 0;
+    ioctl(fd, PERF_EVENT_IOC_DISABLE, 0);
```

Comment:
Why disable the event just before you enable it?

Reply:
I copied this from the other bug checks.

Reply:
FWIW, my implementation never touches enable/disable (even during regular usage), and doesn't even account for any kind of bias in the count read by `rdpmc`, just relies on *always* subtracting two read counts (or comparing them, like here).

But I'm using `rdpmc` with `perf_event_mmap_page`, not reading from the `fd`, so maybe there's some differences there?

Reply:
Ah, we're just doing the disable thing to reset the counter in case code in `start_counter()` accidentally triggers events. So I think this is fine.